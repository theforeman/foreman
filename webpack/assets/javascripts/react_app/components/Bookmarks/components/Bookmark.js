import React from 'react';
import PropTypes from 'prop-types';
import { MenuItem } from 'patternfly-react';
import EllipisWithTooltip from 'react-ellipsis-with-tooltip';

const Bookmark = ({ text, query, onClick }) => (
  <MenuItem onClick={() => onClick(query)}>
    <EllipisWithTooltip>{text}</EllipisWithTooltip>
  </MenuItem>
);

Bookmark.propTypes = {
  onClick: PropTypes.func.isRequired,
  text: PropTypes.string.isRequired,
  query: PropTypes.string.isRequired,
};

export default Bookmark;
