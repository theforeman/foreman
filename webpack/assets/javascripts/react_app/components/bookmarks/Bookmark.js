import React from 'react';
import PropTypes from 'prop-types';
import URI from 'urijs';
import { MenuItem } from 'patternfly-react';
import EllipisWithTooltip from 'react-ellipsis-with-tooltip';

const Bookmark = ({ text, query }) => {
  const handleClick = e => {
    e.preventDefault();
    const uri = new URI(window.location.href);

    uri.setSearch('search', query.trim());
    window.Turbolinks.visit(uri.toString());
  };

  return (
    <MenuItem onClick={handleClick}>
      <EllipisWithTooltip>{text}</EllipisWithTooltip>
    </MenuItem>
  );
};

Bookmark.propTypes = {
  text: PropTypes.string.isRequired,
  query: PropTypes.string.isRequired,
};

export default Bookmark;
