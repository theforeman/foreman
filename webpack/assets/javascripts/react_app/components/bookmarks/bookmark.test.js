import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';
import React from 'react';
import Bookmark from './Bookmark';

function setup() {
  const props = {
    text: 'label',
    query: 'query',
  };

  const wrapper = mount(<Bookmark {...props} />);

  return {
    props,
    wrapper,
  };
}

const setupTurbolinksMock = () => {
  global.Turbolinks = {
    visit: jest.fn(),
  };
};

describe('bookmark', () => {
  it('should create a link to a bookmark', () => {
    setupTurbolinksMock();
    const { wrapper } = setup();

    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.find('a').simulate('click');
    expect(global.Turbolinks.visit).toBeCalled();
    expect(global.Turbolinks.visit).toHaveBeenLastCalledWith(
      'http://localhost/?search=query'
    );
  });
});
