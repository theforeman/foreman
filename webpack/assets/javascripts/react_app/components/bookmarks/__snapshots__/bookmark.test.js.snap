// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bookmark should create a link to a bookmark 1`] = `
<Bookmark
  query="query"
  text="label"
>
  <MenuItem
    bsClass="dropdown"
    disabled={false}
    divider={false}
    header={false}
    onClick={[Function]}
  >
    <li
      className=""
      role="presentation"
    >
      <SafeAnchor
        componentClass="a"
        onClick={[Function]}
        role="menuitem"
        tabIndex="-1"
      >
        <a
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1"
        >
          <EllipisWithTooltip>
            <div
              onMouseEnter={[Function]}
              style={
                Object {
                  "overflow": "hidden",
                  "overflowWrap": "break-word",
                  "textOverflow": "ellipsis",
                  "whiteSpace": "nowrap",
                  "wordBreak": "break-all",
                }
              }
            >
              label
            </div>
          </EllipisWithTooltip>
        </a>
      </SafeAnchor>
    </li>
  </MenuItem>
</Bookmark>
`;
